src/analysis/normed_space/inner_product.lean

```diff
@@ -1374,6 +1382,56 @@ linear_map.mk_continuous
 
 end norm
 
+section bessels_inequality
+
+variables {ι: Type*} (x : E) {v : ι → E}
+
+lemma bessel_finite {s : finset ι} (hv : orthonormal 𝕜 v) : ∑ i in s, ∥ ⟪v i, x⟫ ∥^2 ≤ ∥x∥ ^2 :=
+begin
+  rw ←sub_nonneg,
+  suffices hbf: ∥ x -  ∑ i in s, ⟪v i, x⟫ • (v i) ∥^2 = ∥x∥^2 - ∑ i in s, ∥ ⟪v i, x⟫ ∥^2,
+  { rw ←hbf, simp only [norm_nonneg, pow_nonneg], },
```

Comment:
With some exceptions, the style is to put each tactic application on a new line -- this applies both here and in a few places below.